components/webhook-mgt/org.wso2.carbon.identity.webhook.metadata/src/main/java/org/wso2/carbon/identity/webhook/metadata/internal/dao/impl/WebhookMetadataDAOImpl.java

```diff
@@ -0,0 +1,186 @@
+import org.wso2.carbon.database.utils.jdbc.exceptions.TransactionException;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.organization.resource.sharing.policy.management.constant.PolicyEnum;
+import org.wso2.carbon.identity.webhook.metadata.api.exception.WebhookMetadataException;
+import org.wso2.carbon.identity.webhook.metadata.api.exception.WebhookMetadataServerException;
+import org.wso2.carbon.identity.webhook.metadata.api.model.BinaryObject;
+import org.wso2.carbon.identity.webhook.metadata.api.model.OrganizationPolicy;
+import org.wso2.carbon.identity.webhook.metadata.api.model.WebhookMetadataProperties;
+import org.wso2.carbon.identity.webhook.metadata.api.model.WebhookMetadataProperty;
+import org.wso2.carbon.identity.webhook.metadata.internal.constant.WebhookMetadataSQLConstants;
+import org.wso2.carbon.identity.webhook.metadata.internal.dao.WebhookMetadataDAO;
+
+import java.sql.SQLException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Implementation of the WebhookMetadataDAO interface for managing webhook metadata properties.
+ */
+public class WebhookMetadataDAOImpl implements WebhookMetadataDAO {
+
+    private static final String ORGANIZATION_POLICY_PROPERTY_NAME = "organizationPolicy";
+
+    @Override
+    public WebhookMetadataProperties getWebhookMetadataProperties(int tenantId) throws WebhookMetadataException {
+
+        Map<String, WebhookMetadataProperty> propertiesMap = getWebhookPropertiesFromDB(tenantId);
+
+        WebhookMetadataProperty orgPolicyProperty = propertiesMap.get(ORGANIZATION_POLICY_PROPERTY_NAME);
+        OrganizationPolicy organizationPolicy = null;
+        if (orgPolicyProperty != null && orgPolicyProperty.isPrimitive()) {
+            organizationPolicy =
+                    new OrganizationPolicy(PolicyEnum.getPolicyByPolicyName(orgPolicyProperty.getValue().toString()));
+        }
+
+        return new WebhookMetadataProperties.Builder()
+                .organizationPolicy(organizationPolicy)
+                .build();
+    }
+
+    @Override
+    public void updateWebhookMetadataProperties(WebhookMetadataProperties webhookMetadataProperties, int tenantId)
+            throws WebhookMetadataException {
+
+        OrganizationPolicy updatedOrganizationPolicy = webhookMetadataProperties.getOrganizationPolicy();
+        if (updatedOrganizationPolicy == null) {
+            return;
+        }
+
+        try {
+            updateWebhookPropertiesInDB(
+                    Collections.singletonMap(ORGANIZATION_POLICY_PROPERTY_NAME, new WebhookMetadataProperty.Builder(
+                            updatedOrganizationPolicy.getPolicyCode()).build()), tenantId);
+        } catch (TransactionException e) {
+            throw new WebhookMetadataServerException("Error while updating webhook metadata properties in the system.",
+                    e);
+        }
```

Comment:
IMO ideally for this layer rather policy code should come. It's not the DAOs responsibility to handle logic related to solving references

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/7009/commits/b9f3976655650443f8c8367660276e3171c6a40b